pom.xml

```diff
@@ -0,0 +1,1246 @@
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/surefire-reports</outputDirectory>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>generate-failsafe-html-report</id>
+                        <phase>post-integration-test</phase>
+                        <goals>
+                            <goal>failsafe-report-only</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/failsafe-reports</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <!--JPA API change-->
+        <profile>
+            <id>jpa22</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <persistence.artifactId>jakarta.persistence</persistence.artifactId>
+                <persistence.version>${persistence22.version}</persistence.version>
+            </properties>
+        </profile>
+        <profile>
+            <id>jpa21</id>
+            <properties>
+                <persistence.artifactId>javax.persistence</persistence.artifactId>
+                <persistence.version>${persistence21.version}</persistence.version>
+            </properties>
+        </profile>
+        <!--Testing database profiles-->
+        <!--TODO simplify properties part db.driver.XXXX to load it from file-->
+        <profile>
+            <id>derby</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <test.properties.file>${test.derby.properties.file}</test.properties.file>
+                <test.database>derby</test.database>
+                <!--Used by sql-maven-plugin-->
+                <db.driver.groupId>org.apache.derby</db.driver.groupId>
+                <db.driver.artifactId>derbyclient</db.driver.artifactId>
+                <db.driver.version>${derby.version}</db.driver.version>
+            </properties>
+        </profile>
+        <profile>
+            <id>mysql</id>
+            <properties>
+                <test.properties.file>${test.mysql.properties.file}</test.properties.file>
+                <!--Used by sql-maven-plugin-->
+                <db.driver.groupId>mysql</db.driver.groupId>
+                <db.driver.artifactId>mysql-connector-java</db.driver.artifactId>
+                <db.driver.version>${mysql.version}</db.driver.version>
+                <test.skip.in-memory.db>true</test.skip.in-memory.db>
+            </properties>
+        </profile>
+        <profile>
+            <id>oracle</id>
```

Comment:
this profile should probably define repositories as is described at https://docs.oracle.com/middleware/1213/core/MAVEN/config_maven_repo.htm#MAVEN9014 so one can configure servers in his settings.xml and be able to build specific stuff